strategies/aartur/solver.py

```diff
@@ -0,0 +1,151 @@
+
+    This task can be, for example, collecting ScoreBonus or shooting at enemies.
+
+    Its main method, 'solve', takes AdvancedState and a precalculated Map,
+    and returns BaseMove (which is the most suitable for this task) and 'confidence' in this move.
+    'confidence' is just float, preferably from 0.0 to 1.0, where
+    1.0 means max confidence and 0.0 means totally useless move.
+    """
+
+    def solve(self, state: AdvancedState, mapper: Map) -> tuple[BaseMove, float]:
+        raise NotImplementedError()
+
+
+class CollectBonusSolver(BaseSolver):
+    """Solver for collecting one particular type of Bonus, which is defined in 'bonus_type'.
+    """
+
+    bonus_type: type = None
+
+    def solve(self, state: AdvancedState, mapper: Map) -> tuple[BaseMove, float]:
+        bonuses: list[tuple[int, int]] = []   # list of coordinates of all bonuses on the map
+        for y in range(state.size_y):
+            for x in range(state.size_x):
+                if isinstance(state[y][x], self.bonus_type) and mapper[y][x].visited:
+                    bonuses.append((x, y))
+
+        if not bonuses:
+            return DirectMove(dx=0, dy=0), 0.0
+
+        # choosing the closest bonus as our goal (for now ignoring the bonus value)
+        x, y = min(bonuses, key=lambda elem: mapper[elem[1]][elem[0]].dist)
+        dx, dy = mapper.goto(x, y)
+
+        dist = mapper[y][x].dist
+        confidence = 1.0 - dist * 0.1
+        if self.bonus_type == HealBonus:
+            # confidence is greater if we are low on health
+            # (maybe checking the type of bonus and changing confidence should be somewhere else IDK)
+            confidence *= (1.0 - state.player.health / state.player.max_health)
+
+        return DirectMove(dx=dx, dy=dy), confidence
+
+
+class CollectScoreBonusSolver(CollectBonusSolver):
+    """Solver for collecting ScoreBonus."""
+
+    bonus_type: type = ScoreBonus
+
+
+class CollectHealBonusSolver(CollectBonusSolver):
+    """Solver for collecting HealBonus."""
+
+    bonus_type: type = HealBonus
+
+
+class ShootSolver(BaseSolver):
+    """Solver for shooting at enemy.
+
+    It can return either Shoot, shooting at enemy, or DirectMove, moving to enemy if it is out of range.
+    Prioritizes more damaged and close enemies (close in meaning of moving to them).
+    """
+
+    confidence_moving: float = 0.1  # base confidence if we have to move to the targeted enemy
+    confidence_direct_shooting: float = 0.5  # base confidence if we can shoot enemy directly (which is greater)
+    # except this base confidence, there is other part which depends on how damaged the enemy is
+
+    def solve(self, state: AdvancedState, mapper: Map) -> tuple[BaseMove, float]:
+        # closest positions from which we can shoot at enemy
+        closest_shoot_positions: list[tuple[Player, tuple[int, int]]] = []
+        for enemy in state.other_players:
+            if not enemy.is_alive:
+                # ignoring dead enemies!
```

Comment:
i think it's useless

Reply:
No way i dont how know someone can live without this comment...
Done!